nbextensions/bulkImportCell/main.js

```diff
@@ -0,0 +1,157 @@
+/*
+ * KBase Bulk Import Cell Extension
+ *
+ * Supports kbase bulk import cells
+ *
+ * Note that, out of our control, this operates under the "module as singleton" model.
+ * In this model, the execution of the module the first time per session is the same
+ * as creating a global management object.
+ *
+ * @param {type} $
+ * @param {type} Jupyter
+ * @param {type} html
+ *
+ */
+define([
+    'jquery',
+    'base/js/namespace',
+    'bluebird',
+    'common/runtime',
+    'common/ui',
+    'kb_common/html',
+    './bulkImportCell',
+    'custom/custom'
+], function(
+    $,
+    Jupyter,
+    Promise,
+    Runtime,
+    UI,
+    html,
+    BulkImportCell
+) {
+    'use strict';
+    const t = html.tag,
+        div = t('div'),
+        p = t('p'),
+        CELL_TYPE = 'app-bulk-import';
+
+    /**
+     * This gets called on extension initialization. This iterates over all existing cells,
+     * and if it's a bulk import cell, then we init the module that says so.
+     */
+    function setupNotebook() {
+        return Promise.all(Jupyter.notebook.get_cells().map(function(cell) {
+            if (BulkImportCell.isBulkImportCell(cell)) {
+                try {
+                    const bulkImportCell = new BulkImportCell(cell);
+                }
+                catch(error) {
+
+                    // If we have an error here, there is a serious problem setting up the cell and it is not usable.
+                    // What to do? The safest thing to do is inform the user, and then strip out the cell, leaving
+                    // in it's place a markdown cell with the error info.
+                    // For now, just pop up an error dialog;
+                    var ui = UI.make({
+                        node: document.body
+                    });
+                    ui.showInfoDialog({
+                        title: 'Error',
+                        body: div({
+                            style: {
+                                margin: '10px'
+                            }
+                        }, [
+                            ui.buildPanel({
+                                title: 'Error Starting App Cell',
```

Comment:
bulk import app cell?

Since this is used in other cell types, at some point it'd be good to extract this out and make a generic startup error function.

Reply:
Retitled and extracted.